pkg/pipelines/tasks/tasks_test.go

```diff
@@ -0,0 +1,354 @@
+package tasks
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha2"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGithubStatusTask(t *testing.T) {
+	wantedTask := pipelinev1.Task{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Task",
+			APIVersion: "tekton.dev/v1alpha1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: "create-github-status-task",
+		},
+		Spec: pipelinev1.TaskSpec{
+			Inputs: &pipelinev1.Inputs{
+				Params: []pipelinev1.ParamSpec{
+					pipelinev1.ParamSpec{
```

Comment:
You can use the `createTaskParam*` methods here to shorten the test.

You have assertions that they do what's expected, and it would shorten the test to a more readable length.

You can also use `createEnvFromSecret` later in the test.